src/Symfony/Component/JsonPath/JsonCrawler.php

```diff
@@ -38,36 +38,29 @@ final class JsonCrawler implements JsonCrawlerInterface
         'value' => true,
     ];
 
-    /**
-     * @param resource|string $raw
-     */
-    public function __construct(
-        private readonly mixed $raw,
-    ) {
-        if (!\is_string($raw) && !\is_resource($raw)) {
-            throw new InvalidArgumentException(\sprintf('Expected string or resource, got "%s".', get_debug_type($raw)));
+    public function find(string|JsonPath $query, mixed $data): array
+    {
+        if (!\is_string($data) && !\is_resource($data)) {
+            throw new InvalidArgumentException(\sprintf('Expected string or resource, got "%s".', get_debug_type($data)));
```

Comment:
This exception should be a [`TypeError`](https://www.php.net/manual/en/class.typeerror.php). 
> The argument type being passed to a function does not match its corresponding declared parameter type. 

The type is declared using an annotation.
```suggestion
            throw new \TypeError(\sprintf('Expected string or resource, got "%s".', get_debug_type($data)));
```

Reply:
I updated this part, good point. Thanks!